File: lobwebapp-core/src/main/java/com/espindola/lobwebapp/service/impl/ClientServiceImpl.java
package com.espindola.lobwebapp.service.impl;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.espindola.lobwebapp.domain.Client;
import com.espindola.lobwebapp.repository.contract.ClientRepository;
import com.espindola.lobwebapp.service.contract.ClientService;
import com.espindola.lobwebapp.service.impl.base.AbstractPersonServiceImpl;

@Service
public class ClientServiceImpl extends AbstractPersonServiceImpl<Client> implements ClientService {

	@Autowired
	public ClientServiceImpl(ClientRepository repository) {
		super(repository);
	}
	
}

File: lobwebapp-html/src/script/service/impl/ClientServiceImpl.ts
///<reference path="./../../reference.d.ts"/>
import i0 = require("./base/PersonServiceImpl");

export module service.impl {
    export class ClientServiceImpl extends i0.service.impl.base.PersonServiceImpl<domain.Client> implements d.service.contract.ClientService {

        static $inject = ["$http"];
        constructor($http: ng.IHttpService) {
            super("client", $http);
        }

    }
}

export var register = (moduleName: string) => {
    angular.module(moduleName).lazy.service("ClientService", service.impl.ClientServiceImpl);
};